bootstrap/bootstrap.go

```diff
@@ -93,26 +92,78 @@ func RunAndReturnWaitGroup(
 	var wg sync.WaitGroup
 	deferred := func() {}
 
+	lc := logging.FactoryToStdout(serviceKey)
+
+	// Enforce serviceConfig (which is an interface) is a pointer so we can dereference it later with confidence when required.
+	if reflect.TypeOf(serviceConfig).Kind() != reflect.Ptr {
+		fatalError(fmt.Errorf("serviceConfig parameter must be a pointer to the service's configuration struct"), lc)
+	}
+
 	translateInterruptToCancel(ctx, &wg, cancel)
 
 	configFileName := commonFlags.ConfigFileName()
-	lc := logging.FactoryToStdout(serviceKey)
 
-	// Create new ProviderInfo and initialize it from command-line flag or Environment variable
-	configProviderInfo, err := config.NewProviderInfo(lc, commonFlags.ConfigProviderUrl())
-	if err != nil {
-		fatalError(err, lc)
+	// TODO: remove this check once -r/-registry is back to a bool in release v2.0.0
```

Comment:
Is there a way to do this check under certain versions automatically, so maybe we can avoid to change the code in v2.0.0?

Reply:
We want to change the code for v2.0.0 to clean up this mess caused by the need for backwards compatibility when breaking changes are OK. ;-)

Reply:
ok